### PR TITLE
Updates README per DOC-1495

### PR DESCRIPTION
**Pull request by psoloway**
1. Remove any "what this affects" sections, except where particularly warranted.
2. Make sure that for each parameter, where applicable, there is a data type and a default value.
3. Make sure that for each parameter that's applicable, there is a note if the parameter is optional.
4. Make sure the links in the README work and are accurate.
5. Update the link in the Contributing section to point here: https://docs.puppetlabs.com/forge/contributing.html
6. General copyediting.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,16 +75,68 @@ By default, when the provider triggers a reboot, it will skip any resources in t
...
+The main type of the module, responsible for all its functionality.
+
+####Providers
+
+* `windows`: Default for :kernel => :windows
+* `linux`: Default for :kernel => :linux
+
+####Features
+
+* `supports_reboot_prompting`: Notifies the node's user that a reboot is pending, and let them cancel the pending reboot process. (Available with the `windows` provider. Linux terminal users get a shutdown notice regardless of this parameter, but they can't cancel the reboot.)
```
Comment (elyscape) [inline README.md]: `supports_reboot_prompting` is going away once #62 is merged.

Comment (psoloway) [inline README.md]: @elyscape Do you have an idea of when #62 will be merged? Happy to update my PR, if it's imminent—or I can do another pass on the README once it does go through.

Comment (psoloway) [inline README.md]: Oops, disregard that. I'll go ahead and ask our module engineers.

Comment (elyscape) [inline README.md]: This isn't a particularly big deal, I should note, as if this is merged first then I can just rebase #62 and remove this as part of the change set.

Comment (psoloway) [inline README.md]: @elyscape That sounds like the best approach. That way these PRs can be assessed separately. Thanks!

**Pull request merged**